### PR TITLE
S3 image retrieval route + image display in SinglePost.jsx

### PR DESCRIPTION
**Pull request by zhangbrady4**
https://github.com/CloudClub-uoft/GuessTheLocation/assets/127697715/10dd804d-563f-4e72-8b96-50bff1c0097d

### CODE REVIEW

File: server/routes/POST-upload.js [hunk 1]
```diff
@@ -24,5 +25,6 @@ module.exports = (app, s3Client) => {
         console.log("\nUploaded Object:\n", data);
       }
     });
+	res.send({});
```
Comment (qingyuan-wu) [inline server/routes/POST-upload.js:28]: Can you fix the indent here. Also, can we include a status that indicates the request is successful?

File: server/routes/POST-upload.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 module.exports = (app, s3Client) => {
   app.post("/postupload", (req, res) => {
+    console.log("upload")
     var file;
     var fileExtension;
-    // console.log(req.files);
+    console.log(req.files);
```
Comment (qingyuan-wu) [inline server/routes/POST-upload.js:6]: Let's remove unncessary logging

File: server/routes/GET-image.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      Bucket: process.env.BUCKET_NAME,
+      Key: key,
+    };
+
+    s3Client.getObject(params, (err, data) => {
+      if (err) {
+        console.log(err);
+        res.send({});
+      } else {
+        console.log(`Got image: ${key}`);
```
Comment (qingyuan-wu) [inline server/routes/GET-image.js:17]: remove logging

File: client/src/pages/SinglePost.jsx [hunk 4]
```diff
@@ -1,33 +1,70 @@
...
+import Delete from "../img/temp3.jpg";
+import Menu from "../components/Menu";
+import { Link, useParams } from "react-router-dom";
+import axios from "axios";
 
 const Single = () => {
+  const [loaded, setLoaded] = useState(false); // whether image data has been received from axios
+  const [imageURL, setImageURL] = useState(); // image URL generated from image data
+  const { id } = useParams(); // post id
+  // var filename = "CloudClub.png";  // Hardcoded image key
```
Comment (qingyuan-wu) [inline client/src/pages/SinglePost.jsx:12]: let's remove this

File: client/src/pages/SinglePost.jsx [hunk 5]
```diff
@@ -1,33 +1,70 @@
...
+  useEffect(handleGetImage, []);
+
+  function handleGetImage() {
+    // Call to back-end API
+    axios
+      .get(`/getimage/${id}`)
+      .then((res) => {
+        let imagedata = new Uint8Array(res.data.image.data); // raw data of the image converted to uint8Array
+
+        // let fileExt = res.data.fileExtension;
```
Comment (qingyuan-wu) [inline client/src/pages/SinglePost.jsx:23]: also this

File: client/src/pages/SinglePost.jsx [hunk 6]
```diff
@@ -1,33 +1,70 @@
-import React from 'react'
-import Edit from "../img/temp2.jpg"
-import Delete from "../img/temp3.jpg"
-import Menu from "../components/Menu"
-import { Link } from 'react-router-dom'
+import React, { useEffect, useState } from "react";
+import Edit from "../img/temp2.jpg";
+import Delete from "../img/temp3.jpg";
```
Comment (qingyuan-wu) [inline client/src/pages/SinglePost.jsx:3]: Are these supposed to be buttons? If so, can we use actual HTML button elements?

File: client/src/pages/SinglePost.jsx [hunk 7]
```diff
@@ -1,33 +1,70 @@
...
+        )}
+
+        {/* <img src="https://images.pexels.com/photos/5850083/pexels-photo-5850083.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1" /> */}
         <div className="user">
           <img src="https://images.pexels.com/photos/5876695/pexels-photo-5876695.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1" />
         </div>
-        <div className='info'>
+        <div className="info">
           <span>Celine</span>
           <p>Posted 2 days ago</p>
```
Comment (qingyuan-wu) [inline client/src/pages/SinglePost.jsx:54]: TODO (not for this PR, but future) - don't hard code user and post info (name, time, logo, etc.)

File: client/src/pages/SinglePost.jsx (hunk 4 above)
Comment (zhangbrady4) [reply client/src/pages/SinglePost.jsx:12]: I removed the hardcoded image. The variables in lines 9-11 are needed for the logic to display the image.

File: client/src/pages/SinglePost.jsx (hunk 6 above)
Comment (zhangbrady4) [reply client/src/pages/SinglePost.jsx:3]: Not too sure. These were here when I started working on this file.

**Pull request merged**